File: Seminar5/HomeWork/PresenterStartComplex.java
package HomeWork;

import java.util.logging.*;
import java.util.Scanner;

import HomeWork.Controller.PresenterComplex;
import HomeWork.Model.Complex.DivModelComplex;
import HomeWork.Model.Complex.MultModelComplex;
import HomeWork.Model.Complex.SubModelComplex;
import HomeWork.Model.Complex.SumModelComplex;
import HomeWork.View.Menu;
import HomeWork.View.ViewComplex;

public class PresenterStartComplex {

    PresenterComplex presenterComplex;
    Menu menu;
    Scanner scanner;

    public PresenterStartComplex(Menu menu, Scanner scanner) {
        this.menu = menu;
        this.scanner = scanner;
    }

    public void startComplex() {
        Logger logger = Logger.getGlobal();
        switch (menu.selectFunction()) {
            case 1:
                System.out.println("Какие числа будем суммировать?");
                PresenterComplex p1 = new PresenterComplex(new SumModelComplex(), new ViewComplex());
                p1.buttonClick();
                logger.info("Вычисление прошло успешно");
                break;
            case 2:
                System.out.println("Какие числа будем вычитать?");
                PresenterComplex p2 = new PresenterComplex(new SubModelComplex(), new ViewComplex());
                p2.buttonClick();
                logger.info("Вычисление прошло успешно");
                break;
            case 3:
                System.out.println("Какие числа будем умножать?");
                PresenterComplex p3 = new PresenterComplex(new MultModelComplex(), new ViewComplex());
                p3.buttonClick();
                logger.info("Вычисление прошло успешно");
                break;
            case 4:
                System.out.println("Какие числа будем делить?");
                PresenterComplex p4 = new PresenterComplex(new DivModelComplex(), new ViewComplex());
                p4.buttonClick();
                logger.info("Вычисление прошло успешно");
                break;
            default:
                System.out.println("Неверный ввод");
                logger.info("Ошибка");
                break;
        }

    }

}

File: FinalTaskCalculator/PresenterStartComplex.cs
using System;
using static Controller.PresenterComplex;
using static ModelComplex.DivModelComplex;
using static ModelComplex.MultModelComplex;
using static ModelComplex.SubModelComplex;
using static ModelComplex.SumModelComplex;
using static View.Menu;
using static View.ViewComplex;

public class PresenterStartComplex
{
    View.Menu menu = null!;

    public PresenterStartComplex(View.Menu menu)
    {
        this.menu = menu;
    }

    public void StartComplex()
    {
        switch (menu.SelectFunction())
        {
            case 1:
                Console.WriteLine("Какие числа будем суммировать?");
                Controller.PresenterComplex p1 = new Controller.PresenterComplex(new ModelComplex.SumModelComplex(), new View.ViewComplex());
                p1.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");
                break;
            case 2:
                Console.WriteLine("Какие числа будем вычитать?");
                Controller.PresenterComplex p2 = new Controller.PresenterComplex(new ModelComplex.SubModelComplex(), new View.ViewComplex());
                p2.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");
                break;
            case 3:
                Console.WriteLine("Какие числа будем умножать?");
                Controller.PresenterComplex p3 = new Controller.PresenterComplex(new ModelComplex.MultModelComplex(), new View.ViewComplex());
                p3.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");
                break;
            case 4:
                Console.WriteLine("Какие числа будем делить?");
                Controller.PresenterComplex p4 = new Controller.PresenterComplex(new ModelComplex.DivModelComplex(), new View.ViewComplex());
                p4.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");
                break;
            default:
                Console.WriteLine("Неверный ввод");
                Console.WriteLine("Ошибка");
                break;
        }

    }

}